lib/blacklight/configuration/fields.rb

```diff
@@ -137,17 +120,35 @@ def add_blacklight_field config_key, *args, &block
 
       private
 
+      ##
+      # Using the Luke handler see add any fields in the index that match the field_config
+      def handle_matching_fields(config_key, field_config, &block)
+        salient_fields = luke_fields.select do |k, _v|
+          k =~ field_config.match
+        end
+
+        salient_fields.each_key do |field|
+          config = field_config.dup
+          config.match = nil
+          config.field = field
+          config.key = field
+          if self[config_key.pluralize][config.key]
+            self[config_key.pluralize][config.key] = config.merge(self[config_key.pluralize][config.key])
+          else
+            add_blacklight_field(config_key, config, &block)
+          end
+        end
+      end
+
       def luke_fields
         if @table[:luke_fields] == false
           return nil
         end
 
         @table[:luke_fields] ||= Rails.cache.fetch("blacklight_configuration/admin/luke", expires_in: 1.hour) do
           begin
-            if repository_class <= Blacklight::Solr::Repository
-              repository = repository_class.new(self)
-              repository.send_and_receive('admin/luke', params: { fl: '*', 'json.nl' => 'map' })['fields']
-            end
+            repository = repository_class.new(self)
```

Comment:
Can you safely remove the condition around this?

Reply:
The tests seem to pass.  I can think of no reason why we can't require repository implementations to implement this interface.

Reply:
Looks like @cbeer last touched this code, here: https://github.com/projectblacklight/blacklight/commit/d75ff8594463f3c078ea0b0e51fc48ed15bef70e

Thoughts on that, Chris?

Reply:
I don't think it's a problem; I don't remember what the luke response looks like, but if we think something similar will work cross-repository, I don't see a problem with it.